### ISSUE TITLE
git run manifest not known on podman run 

### ISSUE THREAD

**Issue by powerserg19** [containers/podman#19202]
### Issue Description

Hi All 
When I  build an image tagged ```mycompany.registry/mydir/myapplication:latest``` and  then try to run 
podman run mycompany.registry/mydir/myapplication

I am getting this error
```
Trying to pull mycompany.registry/mydir/myapplication...
Error: initializing source docker:mycompany.registry/mydir/myapplication: reading manifest latest in  mycompany.registry/mydir/myapplication : manifest unknown: The named manifest is not known to the registry.
```
I have to either push the image to the registry or pass it the image ID instead.
Shouldnt podmand check local first before going out to remote registry? 

`Podman version 4.5.1`
`host:
  arch: arm64
  buildahVersion: 1.30.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.7-2.fc38.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 98.16
    systemPercent: 0.96
    userPercent: 0.88
  cpus: 1
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: coreos
    version: "38"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.3.8-200.fc38.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 895926272
  memTotal: 2048483328
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.8.5-1.fc38.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.5
      commit: b6f80f766c9a89eb7b1440c0a70ab287434b17ed
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-12.fc38.aarch64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 2h 11m 12.00s (Approximately 0.08 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 106769133568
  graphRootUsed: 3147145216
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 9
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.5.1
  Built: 1685123899
  BuiltTime: Fri May 26 13:58:19 2023
  GitCommit: ""
  GoVersion: go1.20.4
  Os: linux
  OsArch: linux/arm64
  Version: 4.5.1`

### Steps to reproduce the issue

Steps to reproduce the issue
1.
2.
3.

### Describe the results you received

Describe the results you received

### Describe the results you expected

Describe the results you expected

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by powerserg19** [comment]
I was passing in this in during run but it was not built with it
```--platform linux/amd64```

**Comment by flouthoc** [comment]
@powerserg19 did you specify `--platform linux/amd64` while building the image ? cause your `podman` VM  is runnning on `arm64` not `amd64`.  See output of your `podman info`

Since in case of `podman-remote` the actual image which is built is for `arm64` on the remote side and not on the client side. You can verify by inspecting the image before running it.

In layman terms:
* Your client is amd64
* Podman is running inside VM which is `arm64`
* Podman is building image for host architecture i.e `arm64`.
* While running you are specifiying `amd64` which is not present.

I hope this helps. Please feel free to re-open the issue if something is unresolved.

**Comment by powerserg19** [comment]
@flouthoc  thanks for your clarification.  I had originally specified amd64 to satisfy the warning
`WARNING: image platform (linux/amd64/v8) does not match the expected platform (linux/arm64) `
but it burned me in the long run
